src/HLLD.hpp

```diff
@@ -0,0 +1,343 @@
+#ifndef HLLD_HPP_ // NOLINT
+#define HLLD_HPP_
+
+#include "AMReX_Extension.H"
+#include "AMReX_GpuQualifiers.H"
+#include <AMReX.H>
+#include <AMReX_REAL.H>
+
+#include "ArrayView.hpp"
+#include "HydroState.hpp"
+#include "valarray.hpp"
+
+namespace quokka::Riemann
+{
+#define DELTA 1.0e-4
+
+template <class T> constexpr auto SQUARE(const T x) -> T { return x * x; }
+
+template <int N_scalars> auto FastMagnetosonicSpeed(double gamma, quokka::HydroState<N_scalars> const state, const double bx) -> double
+{
+	double gp = gamma * state.P;
+	double bx_sq = bx * bx;
+	double byz_sq = state.by * state.by + state.bz * state.bz;
+	double b_sq = bx_sq + byz_sq;
+	double bgp_p = b_sq + gp;
+	double bgp_m = b_sq - gp;
+	return std::sqrt(0.5 * (bgp_p + std::sqrt(bgp_m * bgp_m + 4.0 * gp * byz_sq)) / state.rho);
+}
+
+inline auto GetWeightForCT(double dflx, double rho_L, double rho_R, double dx, double dt) -> double
+{
+	double v_over_c = (1024.0) * dt * dflx / (dx * (rho_L + rho_R));
```

Comment:
Do you know where the factor of `1024.0` comes from?

Reply:
Not yet. I will get back to you on this.

Reply:
If you haven't figured this out from reading the code, you could open an issue on the Athena++ Github and ask about this.